xbmc/music/MusicDatabase.cpp

```diff
@@ -981,27 +999,16 @@ bool CMusicDatabase::GetAlbum(int idAlbum, CAlbum& album, bool getSongs /* = tru
     if (idAlbum == -1)
       return false; // not in the database
 
+    //Get album, song and album song info data using separate queries/datasets because we can have 
+    //multiple roles per artist for both album and songs and that makes a single combined join impractical
+    //Get album datae
     std::string sql;
-    if (getSongs)
-    {
-      sql = PrepareSQL("SELECT albumview.*,albumartistview.*,songview.*,songartistview.*,albuminfosong.* "
-                       "  FROM albumview "
-                       "  LEFT JOIN albumartistview ON albumview.idAlbum = albumartistview.idAlbum "
-                       "  JOIN songview ON albumview.idAlbum = songview.idAlbum "
-                       "  LEFT JOIN songartistview ON songview.idSong = songartistview.idSong "
-                       "  LEFT JOIN albuminfosong ON albumview.idAlbum = albuminfosong.idAlbumInfo "
-                       "  WHERE albumview.idAlbum = %ld "
-                       "  ORDER BY albumartistview.iOrder, songview.iTrack, songartistview.iOrder", idAlbum);
-    }
-    else
-    {
-      sql = PrepareSQL("SELECT albumview.*,albumartistview.* "
-                       "  FROM albumview "
-                       "  LEFT JOIN albumartistview ON albumview.idAlbum = albumartistview.idAlbum "
-                       "  WHERE albumview.idAlbum = %ld "
-                       "  ORDER BY albumartistview.iOrder", idAlbum);
-    }
-
+    sql = PrepareSQL("SELECT albumview.*,albumartistview.* "
+                     " FROM albumview "
+                     " LEFT JOIN albumartistview ON albumview.idAlbum = albumartistview.idAlbum "
```

Comment:
It depends if we can have albums without album artists? The original Kodi processing (pre Musicbrainz) will ensure that every album has at least one album artist, if it is varied then it is set to "various artists" etc. However if the album has MBID then this is not enforced. Although Picard tags files with an album artist(s), it is possible for users to mess with this tagging (something that happens even if we don't like it) and remove the album artist. This means we could have albums without artists in real life user music libaries. An outer join here means that we would still see these (slightly messed up) albums.
So what do we do anyone?